synapse/module_api/__init__.py

```diff
@@ -72,6 +72,16 @@ def public_room_list_manager(self):
         """
         return self._public_room_list_manager
 
+    def get_user_by_id(self, user_id: str) -> Optional[Dict[str, Any]]:
+        """Get user by user_id
+
+        Args:
+            user_id: Fully qualified user id.
+        Returns:
+            the user dictionary or None if not found.
+        """
+        return self._store.get_user_by_id(user_id)
```

Comment:
isn't get_user_by_id async?

Reply:
Good call. Is `defer.Deferred` the correct type or is the some other way to signify with typing that it's a deferred containing either a dictionary or None? Looking around, that type seems to be set as is for return values where used.

Reply:
generally it's best to declare the function `async` and put an `await` in here.